File: algorithms/SearchAndSort/MergeSort/mergeSort.py
def mergeSort(sortList):
    if len(sortList) > 1:
        mid = len(sortList)//2
        lefthalf = sortList[:mid]
        righthalf = sortList[mid:]

        mergeSort(lefthalf)
        mergeSort(righthalf)

        i = 0
        j = 0
        k = 0
        print(i < len(lefthalf) and j < len(righthalf))
        while i < len(lefthalf) and j < len(righthalf):
            if lefthalf[i] < righthalf[j]:
                sortList[k] = lefthalf[i]
                i = i + 1
            else:
                sortList[k] = righthalf[j]
                j = j + 1
            k = k + 1

        while i < len(lefthalf):
            sortList[k] = lefthalf[i]
            i = i + 1
            k = k + 1

        while j < len(righthalf):
            sortList[k] = righthalf[j]
            j = j + 1
            k = k + 1

myList = [54,26,93,17,77,31,44,55,20]
mergeSort(myList)
print(myList)

File: algorithms/SearchAndSort/MergeSort/mergeSort.js
function mergeSort(sortList) {
    if (sortList.length > 1) {
        console.log(sortList)
        var mid = Math.round(sortList.length / 2)
        var leftHalf = sortList.slice(0, mid)
        var rightHalf = sortList.slice(mid, sortList.length)

        leftHalf = mergeSort(leftHalf)
        rightHalf = mergeSort(rightHalf)

        var i = 0, j = 0, k = 0;
        
        while (i < leftHalf.length && j < rightHalf.length) {
            if (leftHalf[i] < rightHalf[j]) {
                sortList[k] = leftHalf[i]
                i++
            } else {
                sortList[k] = rightHalf[j]
                j++
            }
            k++
        }

        while (i < leftHalf.length) {
            sortList[k] = leftHalf[i]
            i++
            k++
        }

        while (j < rightHalf.length) {
            sortList[k] = rightHalf[j]
            j++
            k++
        }
    }
    return sortList
}

var myList = [54,26,93,17,77,31,44,55,20]
console.log(mergeSort(myList))